platform/distance.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include "platform/measurement_utils.hpp"
+
+#include <string>
+
+namespace platform
+{
+class Distance
+{
+public:
+  enum class Units
+  {
+    Meters = 1,
+    Kilometers = 2,
+    Feet = 3,
+    Miles = 4,
+
+    // OSM-only
+    NauticalMiles = 5,
+    Inches = 6
+  };
+
+  Distance();
+
+  explicit Distance(double distanceInMeters);
+
+  explicit Distance(double distance, Units units);
+
+  [[nodiscard]] static Distance CreateFormatted(double distanceInMeters);
```

Comment:
Well, I think that [[nodiscard]] here and below is useless and makes only visual trash in code, IMO.

Imagine that we will mark like this all our functions now in the whole code ..

Reply:
[[nodiscard]] shows that it's a bad idea to call the function without storing the result (and using it later)

It was just an experiment and I also don't like how it looks